cmd/gitopsUpdateDeployment_test.go

```diff
@@ -0,0 +1,231 @@
+package cmd
+
+import (
+	"errors"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestBuildRegistryPlusImage(t *testing.T) {
+	t.Parallel()
+	t.Run("build full image", func(t *testing.T) {
+		registryImage, err := buildRegistryPlusImage(&gitopsUpdateDeploymentOptions{
+			ContainerRegistryURL: "https://myregistry.com/registry/containers",
+			ContainerImage:       "myFancyContainer:1337",
+		})
+		assert.NoError(t, err)
+		assert.Equal(t, "myregistry.com/myFancyContainer:1337", registryImage)
+	})
+
+	t.Run("without registry", func(t *testing.T) {
+		registryImage, err := buildRegistryPlusImage(&gitopsUpdateDeploymentOptions{
+			ContainerRegistryURL: "",
+			ContainerImage:       "myFancyContainer:1337",
+		})
+		assert.NoError(t, err)
+		assert.Equal(t, "myFancyContainer:1337", registryImage)
+	})
+	t.Run("without faulty URL", func(t *testing.T) {
+		_, err := buildRegistryPlusImage(&gitopsUpdateDeploymentOptions{
+			ContainerRegistryURL: "//myregistry.com/registry/containers",
+			ContainerImage:       "myFancyContainer:1337",
+		})
+		assert.Error(t, err)
+		assert.EqualError(t, err, "invalid registry url")
+	})
+}
+
+func TestRunGitopsUpdateDeployment(t *testing.T) {
+	t.Parallel()
+	t.Run("successful run", func(t *testing.T) {
+		var configuration = &gitopsUpdateDeploymentOptions{
+			BranchName:           "main",
+			CommitMessage:        "This is the commit message",
+			ServerURL:            "https://github.com",
+			Username:             "admin3",
+			Password:             "validAccessToken",
+			FilePath:             "dir1/dir2/depl.yaml",
+			ContainerName:        "myContainer",
+			ContainerRegistryURL: "https://myregistry.com/registry/containers",
+			ContainerImage:       "myFancyContainer:1337",
+		}
+
+		gitUtilsMock := &validGitUtilsMock{}
+
+		runnerMock := gitOpsExecRunnerMock{}
```

Comment:
There is already `mock.ExecMockRunner{}`, in case you did not notice. It records the calls and you can also specify calls for which it should fail.

Reply:
That's fine, but I need my own for my assertions.